### ISSUE TITLE
Create app option to enable/disable automatic updates (Trac #120)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#124]
Migrated from http://microfluidics.utoronto.ca/microdrop/ticket/120
```json
{
    "status": "closed", 
    "changetime": "2014-04-17T19:39:01", 
    "description": "", 
    "reporter": "ryan", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1397763541728826", 
    "component": "microdrop", 
    "summary": "Create app option to enable/disable automatic updates", 
    "priority": "major", 
    "keywords": "", 
    "version": "0.1", 
    "time": "2012-08-01T17:05:14", 
    "milestone": "Microdrop 1.0", 
    "owner": "cfobel", 
    "type": "enhancement"
}
```

**Comment by cfobel** [comment]
Trac update at 20120801T17:05:27: 
* ryan changed owner from "" to "cfobel"
* ryan changed status from "new" to "assigned"

**Comment by cfobel** [comment]
Trac update at 20120803T02:53:16: 
* christian@fobel.net commented:

> commit 8cd5896f9c09bbdfe9c46d8123c1c973d4f39c96
> Author: Christian Fobel <christian@fobel.net>
> Date:   Thu Aug 2 22:40:09 2012 -0400
> 
>     Add update_automatically app option to plugin m...
>     
>     Add update_automatically app option to PluginManagerController.   When
>     set to True, the plugin server will be be queried and all installed
>     plugins will be updated if there is a newer version on the plugin
>     server.  If update_automatically is set to False, then plugins will
>     only be updated if done so manually in the plugin manager dialog.
>     
>     Fixes #124.


* christian@fobel.net changed resolution from "" to "fixed"
* christian@fobel.net changed status from "assigned" to "closed"

**Comment by cfobel** [comment]
Trac update at 20120808T21:31:23: 
* ryan commented:

> I realize that the title of this ticket is a bit ambiguous. What I had intended was to have an auto-update option that would specify whether or not the application should check for updates at start up. I would prefer that it would still ask the user whether or not to apply each update (or maybe add an option to update all?). Maybe a better wording for this whould be "automatically check for updates".
> 
> Basically, if people want to freeze their setup with a specific application version and set of plugins, I don't want the program to hassle them with new updates every time it starts. For people who choose to turn this option off, we should probably add a menu option to the help menu: "Check for updates".


* ryan changed resolution from "fixed" to ""
* ryan changed status from "closed" to "reopened"

**Comment by cfobel** [comment]
Trac update at 20120809T22:53:26: ryan commented:

> I also think that if this setting is going to apply to both the application and plugins, it should belong to the App class rather than the !PluginManager.

**Comment by cfobel** [comment]
Trac update at 20120811T23:44:33: 
* christian@fobel.net commented:

> commit 58180939c36e7e5e147897adf248595d711bda37
> Merge: c9a92c8adec7bb1c50af2788cd0eb0177d0d2728 4f8653bccc4c5a9f84f01b1e64ac5061b341025b
> Author: Christian Fobel <christian@fobel.net>
> Date:   Sat Aug 11 19:35:55 2012 -0400
> 
>     Merge branch 'auto_updates'
>     
>     This commit fixes #124.


* christian@fobel.net changed resolution from "" to "fixed"
* christian@fobel.net changed status from "reopened" to "closed"

**Comment by cfobel** [comment]
Trac update at 20140417T19:39:01: ryan changed component from "" to "microdrop"
